### PR TITLE
Add more types

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

**Comment by keradus** [comment]
I'm wondering if it's manual effort or automated?

File: src/Linter/ProcessLintingResult.php [hunk 1]
```diff
@@ -23,20 +23,14 @@
...
-     */
-    private $process;
+    private ?string $path;
 
     /**
-     * @var null|string
+     * @var bool
      */
-    private $path;
+    private $isSuccessful;
```
Comment (keradus) [inline src/Linter/ProcessLintingResult.php:33]:
```suggestion
    private bool $isSuccessful;
```

File: src/Runner/FileCachingLintingIterator.php [hunk 2]
```diff
@@ -24,13 +24,13 @@
  */
 final class FileCachingLintingIterator extends \CachingIterator
 {
+    private LinterInterface $linter;
+
     /**
      * @var LintingResultInterface
      */
     private $currentResult;
```
Comment (keradus) [inline src/Runner/FileCachingLintingIterator.php:32]:
```suggestion
    private ?LintingResultInterface $currentResult;
```

Comment (keradus) [reply src/Runner/FileCachingLintingIterator.php:32]: similar for `nextResult`?

File: src/Tokenizer/TokensAnalyzer.php [hunk 3]
```diff
@@ -34,10 +34,7 @@ final class TokensAnalyzer
      */
     private Tokens $tokens;
 
-    /**
-     * @var ?GotoLabelAnalyzer
-     */
-    private $gotoLabelAnalyzer;
+    private ?GotoLabelAnalyzer $gotoLabelAnalyzer = null;
```
Comment (keradus) [inline src/Tokenizer/TokensAnalyzer.php:37]:
```suggestion
    private ?GotoLabelAnalyzer $gotoLabelAnalyzer;
```

**Review by keradus — APPROVED** [review]

File: src/Tokenizer/TokensAnalyzer.php (hunk 3 above)
Comment (SpacePossum) [reply src/Tokenizer/TokensAnalyzer.php:37]: that won't work;
```
Testing PhpCsFixer\Tests\Tokenizer\TokensAnalyzerTest
......SSSSS.....................SSS....E........E.....................E.............SSSSSSSSSSSSSSSSSSSS.S..................................................... 159 / 306 ( 51%)
................................................SSSS...............................................................................................             306 / 306 (100%)

Time: 00:00.153, Memory: 26.50 MB

There were 3 errors:

1) PhpCsFixer\Tests\Tokenizer\TokensAnalyzerTest::testIsConstantInvocation with data set #3 (array(true, true, true), '<?php echo FOO ? BAR : BAZ;')
Error: Typed property PhpCsFixer\Tokenizer\TokensAnalyzer::$gotoLabelAnalyzer must not be accessed before initialization

/home/possum/work/PHP-CS-Fixer/src/Tokenizer/TokensAnalyzer.php:366
/home/possum/work/PHP-CS-Fixer/tests/Tokenizer/TokensAnalyzerTest.php:2281
/home/possum/work/PHP-CS-Fixer/tests/Tokenizer/TokensAnalyzerTest.php:920
```

File: src/Runner/FileCachingLintingIterator.php (hunk 2 above)
Comment (SpacePossum) [reply src/Runner/FileCachingLintingIterator.php:32]: I'm not 100% convinced these PHPDoc are always correct, so I skip it for now as I don't want to propose changes I don't 100% understand

File: src/Linter/ProcessLintingResult.php (hunk 1 above)
Comment (SpacePossum) [reply src/Linter/ProcessLintingResult.php]: thanks added, but with `= null`, see other comment for details

**Pull request closed without merging**